src/mds/MDCache.h

```diff
@@ -1058,14 +1058,14 @@ class MDCache {
   friend class C_MDC_Join;
 
 public:
-  void replicate_dir(CDir *dir, mds_rank_t to, bufferlist& bl);
-  void replicate_dentry(CDentry *dn, mds_rank_t to, bufferlist& bl);
-  void replicate_inode(CInode *in, mds_rank_t to, bufferlist& bl,
+  void encode_replica_dir(CDir *dir, mds_rank_t to, bufferlist& bl);
+  void encode_replica_dentry(CDentry *dn, mds_rank_t to, bufferlist& bl);
+  void encode_replica_inode(CInode *in, mds_rank_t to, bufferlist& bl,
 		       uint64_t features);
   
-  CDir* add_replica_dir(bufferlist::iterator& p, CInode *diri, mds_rank_t from, list<MDSInternalContextBase*>& finished);
-  CDentry *add_replica_dentry(bufferlist::iterator& p, CDir *dir, list<MDSInternalContextBase*>& finished);
-  CInode *add_replica_inode(bufferlist::iterator& p, CDentry *dn, list<MDSInternalContextBase*>& finished);
+  void decode_replica_dir(CDir *&dir, bufferlist::iterator& p, CInode *diri, mds_rank_t from, list<MDSInternalContextBase*>& finished);
+  void decode_replica_dentry(CDentry *&dn, bufferlist::iterator& p, CDir *dir, list<MDSInternalContextBase*>& finished);
+  void decode_replica_inode(CInode *&in, bufferlist::iterator& p, CDentry *dn, list<MDSInternalContextBase*>& finished);
```

Comment:
why does these rename

Reply:
it would look more like a couple, if you change encode function, you always need to do the corresponding change on another part.
also， those function are actually doing the encode and decode things

Reply:
please avoid unnecessary function rename. If you really like to do that, open another PR

Reply:
i only renamed the group of replicate functions,  i just think from now on, all the encode and decode methods are start with prefix "encode" and "decode", `yeah, it's not necessary for this PR, but it would make more sense, what do you think ? .

Reply:
I don't think rename them to encode_replica_foo is better.  these functions do more that than just encoding

Reply:
but, we take an example of CInode replica, below is the code.
  ENCODE_START(1, 1, bl);
  assert(in->is_auth());
  encode(in->inode.ino, bl);  // bleh, minor assymetry here
  encode(in->last, bl);

  //relax locks?
  if (!in->is_replicated()) {
    in->replicate_relax_locks();
  }
  __u32 nonce = in->add_replica(to);
  encode(nonce, bl);

  in->_encode_base(bl, features);
  in->_encode_locks_state_for_replica(bl, mds->get_state() < MDSMap::STATE_ACTIVE);
  ENCODE_FINISH(bl);

----
1. from my view, the code is  about encoding some replica related field of the CInode to a bufferlist and same  for decoding part.
2. it's better for the reader to understand from the function name that  "uh, this function is doing replica encoding/decoding things" 
3. i have added ENCODE_START/DECODE_START to these functions to make it versioned, so, it's more reasonable to rename it to encode_replica_foo/decode_replica_foo
4. original function name  are replicate_inode/add_replica_inode,  now i have changed them to encode_replica_inode/decode_replica_inode , so it reminds developer that encode_replica_foo and decode_replica_foo must be consistent 。 replica_inode/add_replica_inode does not give developer that hint, right ?

those are the reason i do the renaming things here.  of cause, i'm okay with you to change it back if you really think replicate_foo/add_replica_foo is better.

Reply:
how about encode_inodestat, it also contain many other logic, but it gives the reader a hint that the main job of this long function is  about encoding inode fields to a bufferlist and the bufferlist is going to be sent to somewhere.
and another benefit is that, the other developers will know that they need to keep it to do the encoding/decoding job as much as possible, so that we won't got more massive code in these encoding/decoding functions.

Reply:
ping, @ukernel ?